File: Atcoder_#328/javascript/b_11_11.js
function main(input) {
    // 入力を配列に分割
    const lines = input.trim().split('\n');
    const n = parseInt(lines[0]);
    const daysInMonth = lines[1].split(' ').map(Number);

    // ゾロ目の日付のカウント
    let count = 0;

    for (let month = 1; month <= n; month++) {
        for (let day = 1; day <= daysInMonth[month - 1]; day++) {
            const monthStr = month.toString();
            const dayStr = day.toString();

            // 月と日が同じ1桁の数字で構成されているかを確認
            if (monthStr.length === 1 && dayStr.length === 1 && monthStr === dayStr) {
                count++;
            } else if (monthStr.length > 1 && dayStr.length === 1 && new Set(monthStr).size === 1 && monthStr[0] === dayStr) {
                count++;
            } else if (monthStr.length === 1 && dayStr.length > 1 && new Set(dayStr).size === 1 && dayStr[0] === monthStr) {
                count++;
            } else if (monthStr.length > 1 && dayStr.length > 1 && new Set(monthStr).size === 1 && new Set(dayStr).size === 1 && monthStr[0] === dayStr[0]) {
                count++;
            }
        }
    }

    // 結果を出力
    console.log(count);
}

// 標準入力を読み込み、main関数を呼び出し
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf8');
main(input);

File: Atcoder_#328/ruby/b_11_11.rb
n = gets.to_i
days_in_month = gets.split.map(&:to_i)

# ゾロ目の日付のカウント
count = 0

(1..n).each do |month|
  # 各月で日付がゾロ目になるか確認
  (1..days_in_month[month - 1]).each do |day|
    month_str = month.to_s
    day_str = day.to_s
    # 月と日が同じ1桁の数字で構成されているかを確認
    if month_str.length == 1 && day_str.length == 1 && month_str == day_str
      count += 1
    elsif month_str.length > 1 && day_str.length == 1 && month_str.chars.uniq.size == 1 && month_str[0] == day_str
      count += 1
    elsif month_str.length == 1 && day_str.length > 1 && day_str.chars.uniq.size == 1 && day_str[0] == month_str
      count += 1
    elsif month_str.length > 1 && day_str.length > 1 && month_str.chars.uniq.size == 1 && day_str.chars.uniq.size == 1 && month_str[0] == day_str[0]
      count += 1
    end
  end
end

# 結果を出力
puts count
